src/app/review/agree/page.tsx

```diff
@@ -0,0 +1,96 @@
+"use client";
+
+import * as style from "./style.css";
+import { useState } from "react";
+import { useAgreeReviewMutation } from "@app/api/review/agree/hook";
+import { TITLE, CHECKBOX_TEXTS } from "@app/review/agree/constant";
+import Image from "next/image";
+
+import { IcLeftarrow } from "@asset/svg";
+import reviewNoticeFrame from "@asset/image/reviewNoticeFrame.png";
+import danger from "@asset/image/danger.png";
+import { IcCheckbox } from "@asset/svg";
+
+import HeaderNav from "@common/component/HeaderNav/HeaderNav";
+import Divider from "@common/component/Divider/Divider";
+import { Button } from "@common/component/Button";
+
+const page = () => {
+  const [checkedBoxes, setCheckedBoxes] = useState<boolean[]>(Array(4).fill(false));
+  const mutation = useAgreeReviewMutation();
+  // const router = useRouter();
+
+  const handleClickBtn = () => {
+    mutation.mutate();
+    // mutation.mutate(undefined, {
+    //   onSuccess: () => {
+    //     router.push("/review/write");
+    //   },
+    // });
+  };
+
+  const allChecked = checkedBoxes.every((v, i) => i === 0 || v);
+
+  const handleCheckboxToggle = (id: number | "all") => {
+    if (id === "all") {
+      const newValue = !allChecked;
+      setCheckedBoxes(Array(4).fill(newValue));
+    } else {
+      const updated = [...checkedBoxes];
+      updated[id] = !updated[id];
+
+      const subChecks = updated.slice(1);
+      updated[0] = subChecks.every(Boolean);
+
+      setCheckedBoxes(updated);
+    }
+  };
```

Comment:
p3) 함수를 좀 더 나눠보는건 어떤가요? (+ 위에서 정의한 상수값 사용)

```suggestion
  const handleCheckBoxAllToggle = () => {
    const newValue = !allChecked;
    setCheckedBoxes(Array(4).fill(newValue));
  }

  const handleCheckBoxToggle = (idx: number) => {
    const updated = [...checkedBoxes];
    updated[idx] = !updated[idx];

    const subChecks = updated.slice(SUB_CHECKBOX_START_IDX);
    updated[ALL_CHECKBOX_IDX] = subChecks.every(Boolean);

    setCheckedBoxes(updated);
    }
```

Reply:
반영완료했습니다! 감사합니다:)